extensions/amp-redbull-player/0.1/utils/getPlayerParams.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/**
+ * Read elements attributes and convet into a combined string.
+ * @param {!Element} el
+ * @return {string} str
+ */
+const getPlayerParams = el => {
```

Comment:
You don't need this function. Constructing query strings directly without URI-encoding is vvvvv scary, so let's just build a `dict()` to use with `addParamsToUrl`.

You can use [dataset](https://developer.mozilla.org/en-US/docs/Web/API/HTMLOrForeignElement/dataset) and `Object.assign`. It's one line.

In the component implementation just append other parameters explicitly:

```js
// Using data attributes as default parameters.
const params = Object.assign(dict({}), element.dataset);
params['ampTagId'] = this.tagId_;
params['locale'] = this.locale_;

const src = addParamsToUrl(origin, params);
```

Reply:
Actually, you should use `getDataParamsFromAttributes` from `src/dom`. That will give you a `JsonObject` right off the bat:

```js
// Using data attributes as default parameters.
const params = getDataParamsFromAttributes(element);
params['ampTagId'] = this.tagId_;
params['locale'] = this.locale_;

const src = addParamsToUrl(origin, params);
```

Still unsure why we need to add `tagId` and `locale` individually :)

Reply:
updated as you suggested

Reply:
added the data-param- prefix